### PR TITLE
Add credential validation and some fixes

### PR DESCRIPTION
**Pull request by Marx86**
* load annoto in demo mode if clientid is not provided
 * get annoto settings validation
 * Settings position values translation bug
 * Media Details Title bug
 * Setup.py missing data
 * Versioning
 * Add description
 * Set License

### CODE REVIEW

**Comment by HenEytan** [comment]
@Marx86 Please split this PR from one that fix all issues, to PR per resolved issue

File: annoto/annoto.py [hunk 1]
```diff
@@ -100,11 +106,32 @@ def _base_view(self, context=None):
...
-        html = self.resource_string("static/html/annoto.html")
-        frag = Fragment(html.format(self=self))
+        context['error'] = {}
+        if not annoto_auth.get('client_id'):
+            context['error']['type'] = 'warning'
+            context['error']['messages'] = [
+                _('You did not provide annoto credentials. And you view it in demo mode.'),
+                _('Please add "annoto-auth:<CLIENT_ID>:<CLIENT_SECRET>" to "Advanced Settings" > "LTI Passports"'),
+            ]
+        elif not self.JWT_RE.match(annoto_auth.get('client_id')):
```
Comment (genadis) [inline annoto/annoto.py]: @Marx86 Why not use jwt decode to validate the JWT? It will check not only the structure, but the signature as well.

**Comment by genadis** [comment]
@Marx86 Please see my note regarding JWT validation (clientId), other than that, looks good

File: annoto/annoto.py (hunk 1 above)
Comment (Marx86) [reply annoto/annoto.py]: Of course the library has such this method.
But verification requires a public key.

Comment (genadis) [reply annoto/annoto.py]: We use HMAC256 algorithm, so there is no need for keys to verify.

Comment (Marx86) [reply annoto/annoto.py]: But with client_id for *.raccoonganf.com i get the exception:
```jwt.exceptions.DecodeError: Signature verification failed```

Comment (genadis) [reply annoto/annoto.py]: I've edited my previous reply, please read it and let me know. thanks

Comment (Marx86) [reply annoto/annoto.py]: Library has checks payload string for json compliance
https://github.com/jpadilla/pyjwt/blob/master/jwt/api_jwt.py#L96,L101
and it is impossible to disable

If payload is not json, but just a string, then there is no difference between regular expression checking and base64 decoding without verifying the signature.
In both cases, we simply check that something is encoded in base64 and has a structure similar to JWT

Comment (genadis) [reply annoto/annoto.py]: @Marx86 can't you use PyJWS? PyJWT inherits PyJWS and uses it's super method for decoding.
(I'm aware thats regex is similar check, but I prefer standard library over custom regex)
Thanks

Comment (Marx86) [reply annoto/annoto.py]: Ok, I updated the pull request

Comment (genadis) [reply annoto/annoto.py]: thanks, merging

**Pull request merged**